### ISSUE TITLE
Incorrect Try Catch statement for TimeOut in New-PasswordStateResource

### ISSUE THREAD

**Issue by colombeen** [dnewsholme/PasswordState-Management#147]
When you invoke certain API calls you can get very strange error messages that indicate time-outs but should show something different.

Should be something like :
```powershell
Try {
  # Process stuff
} Catch [System.Net.WebException] {
  If ($_.Exception.Status -eq 'Timeout') {
    # Timeout output
  } Else {
    # Other output
  }
}
```

**Comment by colombeen** [comment]
I also don't get why this is only implemented in New-PasswordStateResource, and not in the other *-PasswordStateResource functions

**Comment by colombeen** [comment]
Maybe even push it a bit further and just combine the *-PasswordStateResource functions into **Invoke-PasswordStateAPICall**.
That shouldn't be too hard to implement, but it would create a breaking change.
It would be easier to manage the code.

**Comment by dnewsholme** [comment]
> Maybe even push it a bit further and just combine the *-PasswordStateResource functions into **Invoke-PasswordStateAPICall**. That shouldn't be too hard to implement, but it would create a breaking change. It would be easier to manage the code.

Yeah this is definitely a consideration. It shouldn't be a breaking change since the *-PasswordstateResource functions are private and compatibility can be maintained for all public functions.

**Comment by colombeen** [comment]
Yeah I'm sorry, we split off some time ago and in our case the resource functions are still public
